testing/asserts.ts

```diff
@@ -152,17 +152,7 @@ export function assertFalse(expr: unknown, msg = ""): asserts expr is false {
  * assertEquals<number>(1, 2)
  * ```
  */
-export function assertEquals(
-  actual: unknown,
-  expected: unknown,
-  msg?: string,
-): void;
-export function assertEquals<T>(actual: T, expected: T, msg?: string): void;
-export function assertEquals(
-  actual: unknown,
-  expected: unknown,
-  msg?: string,
-): void {
+export function assertEquals<T>(actual: T, expected: T, msg?: string): void {
```

Comment:
The one really nice thing about typing things this way, is that you get auto-complete on the expected values, which can be especially nice when providing an object. For example:

```ts
assertEquals(actual, {
  someProp: 5,
  // ...auto-complete here...
});
```